Annotations/Classes/Views/HiglightView.swift

```diff
@@ -0,0 +1,187 @@
+    
+    let layer = CAShapeLayer()
+    layer.fillColor = NSColor.color(from: .transparent).cgColor
+    layer.mask = maskLayer
+    
+    return layer
+  }
+  
+  func knobAt(point: PointModel) -> KnobView? {
+    return knobs.first(where: { (knob) -> Bool in
+      return knob.contains(point: point)
+    })
+  }
+  
+  func knobAt(rectPoint: RectPoint) -> KnobView {
+    return knobDict[rectPoint]!
+  }
+  
+  func contains(point: PointModel) -> Bool {
+    return maskPath?.contains(point.cgPoint) ?? false
+  }
+  
+  func addTo(canvas: CanvasView) {
+    canvas.canvasLayer.addSublayer(layer)
+  }
+  
+  func removeFrom(canvas: CanvasView) {
+    layer.removeFromSuperlayer()
+    knobs.forEach { $0.removeFrom(canvas: canvas) }
+  }
+  
+  func dragged(from: PointModel, to: PointModel) {
+    let delta = from.deltaTo(to)
+    state.model = model.copyMoving(delta: delta)
+  }
+  
+  func draggedKnob(_ knob: KnobView, from: PointModel, to: PointModel) {
+    if let rectPoint = (RectPoint.allCases.first { (rectPoint) -> Bool in
+      return knobDict[rectPoint]! === knob
+    }) {
+      let delta = from.deltaTo(to)
+      state.model = model.copyMoving(rectPoint: rectPoint, delta: delta)
+    }
+  }
+  
+  func render(state: HighlightViewState, oldState: HighlightViewState? = nil) {
+    if state.model != oldState?.model {
+      path = type(of: self).createPath(model: model)
+      
+      for rectPoint in RectPoint.allCases {
+        knobAt(rectPoint: rectPoint).state.model = state.model.valueFor(rectPoint: rectPoint)
+      }
+      
+      self.delegate?.highlightView(self, didUpdate: self.model, atIndex: self.modelIndex)
+    }
+    
+    if state.isSelected != oldState?.isSelected {
+      if state.isSelected {
+        knobs.forEach { (knob) in
+          layer.addSublayer(knob.layer)
+        }
+      } else {
+        CATransaction.withoutAnimation {
+          knobs.forEach { (knob) in
+            knob.layer.removeFromSuperlayer()
+          }
+        }
+      }
+    }
+  }
+  
+  func updateColor(_ color: NSColor) {
+    layer.strokeColor = color.cgColor
+    state.model = model.copyWithColor(color: color.annotationModelColor)
+  }
+}
+
+class HighlightViewClass: HighlightView {
+  var state: HighlightViewState {
+    didSet {
+      self.render(state: self.state, oldState: oldValue)
+    }
+  }
+  
+  var delegate: HighlightViewDelegate?
+  
+  var layer: CAShapeLayer
+  var maskPath: CGPath?
+  var globalIndex: Int
+  var modelIndex: Int
+  
+  var color: NSColor? {
+    guard let color = layer.strokeColor else { return nil }
+    return NSColor(cgColor: color)
+  }
+  
+  lazy var knobDict: [RectPoint: KnobView] = [
+    .origin: KnobViewClass(model: model.origin),
+    .to: KnobViewClass(model: model.to),
+    .originY: KnobViewClass(model: model.origin.returnPointModel(dx:model.origin.x, dy:model.to.y)),
+    .toX: KnobViewClass(model: model.to.returnPointModel(dx:model.to.x, dy:model.origin.y))
```

Comment:
@ulian-onua the same case with `dx:`, and `dy:`. Should add a space between ":" and passed value.